lib/redis/connection/memory.rb

```diff
@@ -955,6 +955,38 @@ def zrange(key, start, stop, with_scores = nil)
         results[start..stop].flatten.map(&:to_s)
       end
 
+      def zrangebylex(key, start, stop, *opts)
+        data_type_check(key, ZSet)
+        return [] unless data[key]
+        zset = data[key]
+
+        sorted = if zset.identical_scores?
+          zset.keys.sort { |x, y| x.to_s <=> y.to_s }
+        else
+          zset.keys
+        end
+
+        range = get_range start, stop, sorted.first, sorted.last
+
+        filtered = []
+        sorted.each do |element|
```

Comment:
@larrylv 
I don't think `#bsearch` is suitable there. I don't find border elements there, I check each element for range condition.

I handle begin and end positions there
https://github.com/guilleiguaran/fakeredis/pull/135/files#diff-dd66f1d97e61bf36103779390dbe90c9R975